cmd/rook/ceph/cleanup.go

```diff
@@ -58,21 +62,61 @@ func startCleanUp(cmd *cobra.Command, args []string) error {
 			logger.Infof("successfully cleaned up %q directory", cleanupDirPath)
 		}
 		// Remove all the mon directories.
-		monDirs, err := filepath.Glob(path.Join(dataDirHostPath, "mon-*"))
+		if err := cleanMonDir(); err != nil {
+			logger.Errorf("failed to clean up the mon directories in the dataDirHostPath %q", dataDirHostPath)
+		}
+	}
+
+	return nil
+}
+
+func cleanMonDir() error {
+	monDirs, err := filepath.Glob(path.Join(dataDirHostPath, "mon-*"))
+	if err != nil {
+		return errors.Wrapf(err, "failed to find the mon directories on the dataDirHostPath %q", dataDirHostPath)
+	}
+
+	if len(monDirs) == 0 {
+		logger.Infof("no mon directories are available for clean up in the dataDirHostPath %q", dataDirHostPath)
+		return nil
+	}
+
+	for _, monDir := range monDirs {
+		// Clean up mon directory only if mon secret matches with that in the keyring file.
+		deleteMonDir, err := secretKeyMatch(monDir, monSecret)
 		if err != nil {
-			return errors.Wrapf(err, "failed to clean up mon directories on dataDirHostPath %q", dataDirHostPath)
+			logger.Errorf("failed to clean up the mon directory %q on the dataDirHostPath %q. %v", monDir, dataDirHostPath, err)
+			continue
 		}
-		if len(monDirs) > 0 {
-			for _, monDir := range monDirs {
-				if err := os.RemoveAll(monDir); err != nil {
-					logger.Errorf("failed to clean up mon directory %q on dataDirHostPath. %v", monDir, err)
-				} else {
-					logger.Infof("successfully cleaned up mon directory %q on dataDirHostPath", monDir)
-				}
+		if deleteMonDir {
+			if err := os.RemoveAll(monDir); err != nil {
+				logger.Errorf("failed to clean up the mon directory %q on the dataDirHostPath %q. %v", monDir, dataDirHostPath, err)
+			} else {
+				logger.Infof("successfully cleaned up the mon directory %q on the dataDirHostPath", monDir)
 			}
-			logger.Info("completed clean up of the mon directories in the dataDirHostPath")
+		} else {
+			logger.Infof("skipped clean up of the mon directory %q as the secret key did not match", monDir)
 		}
 	}
 
+	logger.Info("completed clean up of the mon directories in the dataDirHostPath")
```

Comment:
I think this message can be removed since we log every condition above already.

Reply:
removed.